packages/zosjobs/src/SubmitJobs.ts

```diff
@@ -248,14 +247,30 @@ export class SubmitJobs {
             });
             return activeJob;
         }
-        // if viewAppSpoolContent option passed, it waits till job status is output
-        // then get content of each spool file and return array of ISpoolFiles object
-        if (parms.viewAllSpoolContent || parms.waitForOutput) {
+        // wait for job status to be OUTPUT for all other situations
+        if (parms.task != null) {
+            parms.task.statusMessage = "Waiting for " + responseJobInfo.jobid + " to enter OUTPUT";
+            parms.task.percentComplete = TaskProgress.THIRTY_PERCENT;
+        }
+        const job: IJob = await MonitorJobs.waitForJobOutputStatus(session, responseJobInfo);
```

Comment:
by removing the `waitForOutput` check, we are forcing every submit command to wait for the output regardless of the options specified (even an explicit `--no-wfo`).
This might be a breaking change as people may be expecting the command to return.
```powershell
> # This is my CLI installed from public NPM
> zowe jobs sub ds 'USERNAME.PUBLIC.CNTL(sleep)' --no-wfo
jobid:   JOB03187
retcode: null
jobname: JOBNAME
status:  INPUT

> # This is my compiled version
> zdev jobs sub ds 'USERNAME.PUBLIC.CNTL(sleep)' --no-wfo
 ███_______| 30%  O | Waiting for JOB03188 to enter OUTPUT
 ```

Reply:
Apologies for duplicating comments.
This is very similar to Timothy's suggestions.